server/spec/product_versioning_spec.rb

```diff
@@ -0,0 +1,248 @@
+    prod3["uuid"].should_not == prod2["uuid"]
+
+    prods = @cp.list_products_by_owner(owner1["key"])
+    prods.size.should == 1
+    prods[0]["uuid"].should == prod3["uuid"]
+    prods[0]["uuid"].should_not == prod1["uuid"]
+    prods[0]["uuid"].should_not == prod2["uuid"]
+
+    prods = @cp.list_products_by_owner(owner2["key"])
+    prods.size.should == 1
+    prods[0]["uuid"].should == prod2["uuid"]
+    prods[0]["uuid"].should_not == prod3["uuid"]
+  end
+
+  it "creates new products when updating content used by shared products" do
+    owner1 = create_owner random_string('test_owner')
+    owner2 = create_owner random_string('test_owner')
+
+    id = random_string("product")
+    name = "shared_product"
+
+    updated_upstream = Date.today
+
+    prod1 = @cp.create_product(owner1["key"], id, name)
+    prod2 = @cp.create_product(owner2["key"], id, name)
+
+    prod1["uuid"].should == prod2["uuid"]
+
+
+    content_id = random_string("content")
+    content_name = "shared_content"
+    content_label = "shared content"
+    content_type = "shared_content_type"
+    content_vendor = "generous vendor"
+
+    content1 = @cp.create_content(
+        owner1["key"], content_name, content_id, content_label, content_type, content_vendor
+    )
+    content2 = @cp.create_content(
+        owner2["key"], content_name, content_id, content_label, content_type, content_vendor
+    )
+
+    prod3 = @cp.add_content_to_product(owner1["key"], id, content_id)
+    prod4 = @cp.add_content_to_product(owner2["key"], id, content_id)
+    prod3["uuid"].should_not == prod1["uuid"]
+    prod3["uuid"].should_not == prod2["uuid"]
+    prod4["uuid"].should_not == prod1["uuid"]
+    prod4["uuid"].should_not == prod2["uuid"]
+    prod3["uuid"].should == prod4["uuid"]
+
+    prods = @cp.list_products_by_owner(owner1["key"])
+    prods.size.should == 1
+    prods[0]["uuid"].should == prod3["uuid"]
+    prods[0]["uuid"].should_not == prod1["uuid"]
+    prods[0]["uuid"].should_not == prod2["uuid"]
+
+    prods = @cp.list_products_by_owner(owner2["key"])
+    prods.size.should == 1
+    prods[0]["uuid"].should == prod4["uuid"]
+    prods[0]["uuid"].should_not == prod1["uuid"]
+    prods[0]["uuid"].should_not == prod2["uuid"]
+
+    # Actual test starts here
+    content3 = @cp.update_content(owner1["key"], content_id, {:label => "new label"})
+
+    prods = @cp.list_products_by_owner(owner1["key"])
+    prods.size.should == 1
+    prods[0]["uuid"].should_not == prod1["uuid"]
+    prods[0]["uuid"].should_not == prod2["uuid"]
+    prods[0]["uuid"].should_not == prod3["uuid"]
+    prods[0]["uuid"].should_not == prod4["uuid"]
+
+    prods = @cp.list_products_by_owner(owner2["key"])
+    prods.size.should == 1
+    prods[0]["uuid"].should == prod4["uuid"]
+    prods[0]["uuid"].should_not == prod1["uuid"]
+    prods[0]["uuid"].should_not == prod2["uuid"]
+  end
+
+  # TODO ? :
+  # - Updating a shared product used by another resource properly links the new product
```

Comment:
Kind of. It's something that's implicitly tested with the content changes, so I don't believe it's entirely dire to get it in. If we're going for 100% coverage though, I probably should.